src/helpers/helpers.ts

```diff
@@ -111,6 +112,50 @@ const normalizeHosts = (data?: string) =>
     ?.split(',')
     ?.filter(Boolean);
 
+/**
+ * Validates hosts textarea content.
+ *
+ * @param {string} data - host textarea content.
+ * @param {number} maxHosts - maximum number of hosts that the field should allow.
+ * @returns ValidatedOptions
+ */
+const validateHosts = (data: string | undefined, maxHosts: number) => {
+  const hostValid = (value: string) => {
+    // maximum hostname length is 253 ASCII characters, but we need some extra space for ranges
+    if (value.length > 300) {
+      return false;
+    }
+
+    const ipRegex = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(\/\d{1,2})?$/;
+    const hostRegex = /[a-zA-Z0-9][a-zA-Z0-9-_.]*/;
+    const hostNumericRangeRegex = /[a-zA-Z0-9][a-zA-Z0-9-_.]*\[[0-9]+:[0-9]+\]*[a-zA-Z0-9-_.]*/;
+    const hostAlphaRangeRegex = /[a-zA-Z0-9][a-zA-Z0-9-_.]*\[[a-zA-Z]{1}:[a-zA-Z]{1}\][a-zA-Z0-9-_.]*/;
+
+    const allRegexes = [ipRegex, hostRegex, hostNumericRangeRegex, hostAlphaRangeRegex];
+    return allRegexes.some(re => new RegExp(re).test(value));
+  };
+
+  if (data === undefined) {
+    return ValidatedOptions.default;
+  }
+
+  const normalizedHosts = normalizeHosts(data);
+
+  if (!normalizedHosts!.length) {
```

Comment:
looks like we forgot the eslint rule `'@typescript-eslint/no-non-null-assertion'` that's on me... apologies. it's difficult to read and there's the baked in JS optional chaining

**required:** all instances of this "non-null-assertion", `!.` should be replaced... prefer optional chaining, `?.`, instead. that would make this line

```
if (!normalizedHosts?.length)
```
and of course update the remaining instances of this

**optional**
add this rule to the eslint config to avoid this scenario again
```
'@typescript-eslint/no-non-null-assertion': 2,
```

Reply:
Done in c8a96d131f94ebd901a405ef37d2ca19ebca73a5 , along with optional.

I didn't like this `variable!`, and felt it is not necessary. In cbfbcd6faa3b010f2204e122283e436d06a328ee I took another look and modified `normalizeHosts` in a way that makes both `?.` and `!.` not necessary.

This is another commit so we can easily drop it if there are any problems with that approach.